futures-util/src/future/shared.rs

```diff
@@ -19,75 +19,101 @@
 //! # }
```

Comment:
This doc test needs to be updated. Just use `future::ready(6)` and `assert_eq!(6, *block_on(shared1))` (no `unwrap`)

Reply:
Done.